cs2pr

```diff
@@ -18,14 +18,43 @@ gc_disable();
 
 $version = '1.0.4';
 
-if ($argc === 1) {
+// options
+$colorize = false;
+$gracefulWarnings = false;
+
+// parameters
+$params = [];
+foreach ($argv as $arg) {
+    if (substr($arg, 0, 2) === '--') {
+        $option = substr($arg, 2);
+        switch ($option) {
+        case 'graceful-warnings':
+            $gracefulWarnings = true;
+            break;
+        case 'colorize':
+            $colorize = true;
+            break;
+        default:
+            echo "Unknown option ".$option."\n";
+            exit(9);
+        }
+    } else {
+        $params[] = $arg;
+    }
+}
+
+if (count($params) === 1) {
     $xml = stream_get_contents(STDIN);
-} elseif ($argc === 2 && file_exists($argv[1])) {
-    $xml = file_get_contents($argv[1]);
+} elseif (count($params) === 2 && file_exists($params[1])) {
+    $xml = file_get_contents($params[1]);
 } else {
     echo "cs2pr $version\n";
     echo "Annotate a Github Pull Request based on a Checkstyle XML-report.\n";
-    echo "Usage: ". $argv[0] ." <filename>\n";
+    echo "Usage: ". $params[0] ." [OPTION]... <filename>\n";
+    echo "\n";
+    echo "Option can be:\n";
+    echo "  --graceful-warnings   Don't exit with error codes if there are only warnings.\n";
+    echo "  --colorize            Colorize the output (still compatible with Github Annotations)\n";
```

Comment:
Would it make sense to auto activate the colorize in GithubAction jobs? Would it make a difference?

Reply:
There is no advantage for Github Action jobs, but on the other hand, there is also no disadvantage (you won't see a difference). However, making this option activated by default would be beneficial for command line.

Reply:
Will keep it disabled per default for now